eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -114,7 +114,10 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
         case HtlcResult.OnChainFail(cause) =>
           // if the outgoing htlc is being resolved on chain, we treat it like a local error but we cannot retry
           handleLocalFail(d, cause, isFatal = true)
-        case HtlcResult.Disconnected(_) =>
+        case HtlcResult.ChannelFailureBeforeSigned =>
+          // if the outgoing htlc is being resolved on chain, we treat it like a local error but we cannot retry
+          handleLocalFail(d, ChannelFailureException, isFatal = true)
```

Comment:
This one isn't fatal, we can retry with another channel since we don't need to wait to settle the htlc on-chain.

Reply:
Good catch!

Reply:
Done in c6ed34f3d.